internal/pubsub/samples/pub/gcp.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	raw "cloud.google.com/go/pubsub/apiv1"
+	"gocloud.dev/gcp"
+	"gocloud.dev/internal/pubsub"
+	"gocloud.dev/internal/pubsub/gcppubsub"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	grpccreds "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+)
+
+// openGCPTopic returns the GCP topic based on the topic ID.
+func openGCPTopic(ctx context.Context, topicID string) (*pubsub.Topic, func(), error) {
+	pubClient, cleanup, err := openGCPPubClient(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	proj, err := getGCPProjectID()
```

Comment:
This could be combined with @vangent's suggestion of using URLs, so everything including the project ID would go into a single parameter. Then we could write things like
```
$ t=gcppubsub://projects/go-cloud-test-216917/topics/test-topic
$ s=gcppubsub://projects/go-cloud-test-216917/subscriptions/test-subscription-1
$ echo foo | gcmsg pub $t
$ gcmsg sub $s
```

Reply:
Done.